code/datums/martial/sleeping_carp.dm

```diff
@@ -104,31 +105,30 @@
 			D.death() //FINISH HIM!
 		D.apply_damage(50, A.dna.species.attack_type, BODY_ZONE_CHEST)
 		playsound(get_turf(D), 'sound/weapons/punch1.ogg', 75, 1, -1)
-		return 1
+		return TRUE
 	return basic_hit(A,D)
 
 /datum/martial_art/the_sleeping_carp/grab_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
 	if(A==D)
-		return 0 //prevents grabbing yourself
+		return FALSE //prevents grabbing yourself
 	if(A.a_intent == INTENT_GRAB)
 		add_to_streak("G",D)
 		if(check_streak(A,D)) //doing combos is prioritized over upgrading grabs
-			return 1
+			return TRUE
+		old_grab_state = A.grab_state 
 		D.grabbedby(A, 1)
-		if(A.grab_state == GRAB_PASSIVE)
+		if(old_grab_state == GRAB_PASSIVE)
 			D.drop_all_held_items()
 			A.grab_state = GRAB_AGGRESSIVE //Instant agressive grab if on grab intent
 			log_combat(A, D, "grabbed", addition="aggressively")
 			D.visible_message("<span class='warning'>[A] violently grabs [D]!</span>", \
 								"<span class='userdanger'>[A] violently grabs you!</span>")
-	else
-		D.grabbedby(A, 1)
```

Comment:
did you mean to remove this?

Reply:
yes ty

Reply:
if grab_act doesnt return TRUE living.dm has an else statement that does D.grabbedby(A, 1) automatically